ycmd/completers/all/identifier_completer.py

```diff
@@ -37,6 +37,10 @@ def __init__( self, user_options ):
     self._completer = ycm_core.IdentifierCompleter()
     self._tags_file_last_mtime = defaultdict( int )
     self._logger = logging.getLogger( __name__ )
+    try:
+      self._max_candidates = user_options[ 'max_num_identifier_candidates' ]
+    except KeyError:
+      self._max_candidates = MAX_IDENTIFIER_COMPLETIONS_RETURNED
```

Comment:
Oh sorry, I didn't understand what you were saying. Anyway the same argument could be done for every other option. Let's say that you provide an option file without the `min_num_of_chars_for_completion` entry, [ycmd/completers/completer.py#L104](https://github.com/Valloric/ycmd/blob/44eb4008ad17b48c7f1a1d08a52c6dc2b21f59f1/ycmd/completers/completer.py#L104) we pass the `user_options` that is basically what was inside in the option file and no check was done. Sorry if I still didn't understand and this was all noise.